packages/react-pose/src/components/Transition/children.ts

```diff
@@ -1,58 +1,69 @@
 import * as React from 'react';
 import { ReactElement } from 'react';
-import { CurrentPose } from '../PoseElement/types';
-import { MergeChildrenProps, Props } from './types';
+import { Props, State } from './types';
+import { invariant } from 'hey-listen';
 const { Children, cloneElement } = React;
 
-const getKey = (child: ReactElement<any>): string => child.key as string;
-
-const filterChildProps = ({ children, _pose, onPoseComplete, popFromFlow, ...props }: Props) => props;
+const filterChildProps = ({
+  children,
+  _pose,
+  onPoseComplete,
+  popFromFlow,
+  ...props
+}: Props) => props;
+
+const getKey = (child: ReactElement<any>): string => {
+  invariant(
+    child && child.key !== null,
+    'Every child of Transition must be given a unique key'
+  );
 
-const animateChildrenList = (
-  incomingChildren: Array<ReactElement<any>>,
-  pose: CurrentPose,
-  initialPose?: CurrentPose
-) => {
-  const children: Array<ReactElement<any>> = [];
+  const childKey =
+    typeof child.key === 'number' ? child.key.toString() : child.key;
 
-  Children.forEach(incomingChildren, (child: ReactElement<any>) =>
-    children.push(cloneElement(child, { pose, initialPose }))
-  );
-  return children;
+  return childKey.replace('.$', '');
```

Comment:
It might be a holdover from React Native but over there the `key` sporadically seems to be `'.$'`, I haven't figured out why yet but seems to be after using `Children.isArray`

Reply:
FYI https://reactjs.org/docs/react-api.html#reactchildren

> React.Children.toArray() changes keys to preserve the semantics of nested arrays when flattening lists of children. That is, toArray prefixes each key in the returned array so that each element’s key is scoped to the input array containing it.

You should probably document that you do this normalization somewhere - When I create a `<PoseGroup>` I map children that have gone through `toArray()` like so `<AnimatedDiv key={`wrapper-${child.key}`}>`  (AnimatedDiv being the pose component) and the above normalization broke my code, easy fix to simply remove '.$' from my key as well but still.

Reply:
I think I understand your issue, but could you prepare a simple codesandbox showcasing the problem? It would help me to fix it in react-pose itself and write a test that would prevent regressions.